controller/controller.go

```diff
@@ -99,39 +100,111 @@ func (c *Controller) Attach(attachRequest map[string]string) resources.FlexVolum
 	_, err := c.Client.GetVolume(getVolumeRequest)
 
 	if err != nil {
-		return resources.FlexVolumeResponse{
+		return k8sresources.FlexVolumeResponse{
 			Status:  "Failure",
 			Message: "Failed checking volume, call create before attach",
 			Device:  volumeName}
 
 	}
 
-	return resources.FlexVolumeResponse{
+	return k8sresources.FlexVolumeResponse{
 		Status:  "Success",
 		Message: "Volume already attached",
 		Device:  volumeName,
 	}
 
 }
 
+//GetVolumeName checks if volume is attached
+func (c *Controller) GetVolumeName(getVolumeNameRequest k8sresources.FlexVolumeGetVolumeNameRequest) k8sresources.FlexVolumeResponse {
+	c.logger.Println("controller-isAttached-start")
+	defer c.logger.Println("controller-isAttached-end")
+	volumeName, ok := getVolumeNameRequest.Opts["volumeName"]
+	if !ok {
+		return k8sresources.FlexVolumeResponse{
+			Status:  "Failure",
+			Message: fmt.Sprintf("Failed checking volumeName, ", getVolumeNameRequest.Opts),
+		}
+
+	}
+	return k8sresources.FlexVolumeResponse{
```

Comment:
I don't understand how you know that the volume exist?  i don't see any call to ubiquity server to validate if the vol exist.  What do I miss here?